pkg/server/biz/accelerator/cleaner/clean.go

```diff
@@ -0,0 +1,249 @@
+							corev1.ResourceCPU:    resource.MustParse(getOrDefault(c.config, corev1.ResourceRequestsCPU, "50m")),
+							corev1.ResourceMemory: resource.MustParse(getOrDefault(c.config, corev1.ResourceRequestsMemory, "32Mi")),
+						},
+						Limits: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse(getOrDefault(c.config, corev1.ResourceLimitsCPU, "100m")),
+							corev1.ResourceMemory: resource.MustParse(getOrDefault(c.config, corev1.ResourceLimitsMemory, "128Mi")),
+						},
+					},
+				},
+			},
+			Volumes: []corev1.Volume{
+				{
+					Name: common.DefaultCacheVolumeName,
+					VolumeSource: corev1.VolumeSource{
+						PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+							ClaimName: pvcName,
+							ReadOnly:  false,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	pod, err := c.clusterClient.CoreV1().Pods(pvcNamespace).Create(cacheCleanPod)
+	if err != nil {
+		return nil, err
+	}
+
+	go c.watch(pvcNamespace, pod.Name)
+
+	startTime := pod.DeepCopy().CreationTimestamp
+	status := &v1alpha1.AccelerationCacheCleanupStatus{
+		TaskID:             pod.Name,
+		Phase:              "Running",
+		StartTime:          startTime,
+		LastTransitionTime: startTime,
+	}
+	if err := c.writeDownResult(status); err != nil {
+		log.Warningf("Write cache clean resutl error: %v", err)
+		return status, err
+	}
+	return status, nil
+}
+
+func (c *Cleaner) watch(namespace, podName string) {
+	var statusToUpdate *v1alpha1.AccelerationCacheCleanupStatus
+
+	defer func(ns, name string) {
+		if err := c.writeDownResult(statusToUpdate); err != nil {
+			log.Errorf("write down result error: %v, result: %v", err, statusToUpdate)
+		}
+		if err := c.stopClean(ns, name); err != nil {
+			log.Errorf("Stop cache cleaner pod %s/%s error: %v", ns, name, err)
+		}
+	}(namespace, podName)
+
+	w, err := c.clusterClient.CoreV1().Pods(namespace).Watch(metav1.ListOptions{
+		FieldSelector: fmt.Sprintf("metadata.name=%s", podName),
+	})
+	if err != nil {
+		log.Warningf("Watch cache cleanup, start to watch pod failed, namespace: %v, pod name: %v", namespace, podName)
+		return
+	}
+
+	defer w.Stop()
+
+	for {
+		for e := range w.ResultChan() {
+			pod, ok := e.Object.(*corev1.Pod)
+			if !ok {
+				log.Warningf("Watch cache cleanup, object is not a pod, event type: %v, event object: %v", e.Type, e.Object)
+			}
+
+			switch pod.Status.Phase {
+			case corev1.PodSucceeded:
+				statusToUpdate = &v1alpha1.AccelerationCacheCleanupStatus{
+					TaskID:             pod.Name,
+					Phase:              "Succeeded",
+					StartTime:          pod.DeepCopy().CreationTimestamp,
+					LastTransitionTime: metav1.NewTime(time.Now()),
+				}
+				if len(pod.Status.ContainerStatuses) > 0 && pod.Status.ContainerStatuses[0].Name == common.CacheCleanupContainerName {
+					statusToUpdate.LastTransitionTime = pod.Status.ContainerStatuses[0].State.Terminated.FinishedAt
+				}
+				return
+			case corev1.PodRunning, corev1.PodPending:
```

Comment:
This case is not needed as it is covered by `default`.

Reply:
If the pod status is running or pending, we can continue the loop without doing anything.